### PR TITLE
MattT/APPEALS-9224: Standardize Labels and Spacing in Pre-Docket Case Timeline Entries

### PR DESCRIPTION
**Pull request by ThorntonMatthew**
Resolves [APPEALS-9224](https://vajira.max.gov/browse/APPEALS-9224)

## Description
For the sake of uniformity, we would like for all case timeline sections to follow the same formatting guidelines that have been outlined in the acceptance criteria.

## Acceptance Criteria
- [ ] The order that entries are listed in should be, from top-to-bottom, most recent to least recent.
- [ ] If the timeline section is representing information from multiple modal fields, they should be labeled.
   - [ ] Labels, including the terminating colon, should be bolded.
   - [ ] The text that is being labeled (the inputs from the modal) should be on a new line directly below the label. Ex: **My Label:**\nThe text from the modal's form field.
- [ ] Timeline sections that only represent the information from a single modal field should not have a label present.

## Testing Plan
0. Prepare a medley of appeals for the various scenarios that this testing plan will cover:
```ruby
10.times do
  appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
  parent_task = appeal.tasks.last
  emo_document_search_task=EducationDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: EducationEmo.singleton )
end

10.times do
  appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
  parent_task = appeal.tasks.last
  vha_document_search_task=VhaDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: VhaCamo.singleton )
end

5.times do
  appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
  parent_task = appeal.tasks.last
  vha_document_search_task=VhaDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: VhaCaregiverSupport.singleton )
end
```

### VHA - Caregiver Scenarios

1. Log into Caseflow as a CSP user (ex: `CAREGIVERUSER`) and navigate to the VHA CSP queue. [Local Link](http://localhost:3000/organizations/vha-csp)

2. Go to the Case Details page for one of the appeals. Select "Return to Board Intake" from the actions dropdown. Select a reason for return and enter text in the textarea field. Submit the modal. Navigate back to the Case Details page and review the "Task Instructions" entry in the case timeline.

3. Go to the Case Details page for another one of the appeals. Select "Documents ready for Board Intake review" from the actions dropdown. Select the location of the documents (besides "Other") and enter text into the text box. Navigate back to the Case Details page and review the "Task Instructions" entry in the case timeline.

4. Repeat step 3 for another appeal. This time select "Other" and fill in the additional textbox.

### VHA - CAMO Scenarios

1. You will need to interact with appeals are three users for these scenarios. I've been able to log into all three at once by have a Chrome window, a Chrome incognito window, and a Microsoft Edge window open. You can of course just switch users in a single window as needed. You will need to log in as the following users:

VHA CAMO User: `CAMOUSER`
VHA PO User: `VHAPOUSER`
VHA Regional Office/VISN User: `VISNUSER`

Their queues:
[VHA CAMO Queue](http://localhost:3000/organizations/vha-camo)
[VHA PO Queue](http://localhost:3000/organizations/community-care-payment-operations-management)
[VHA VISN Queue](http://localhost:3000/organizations/va-new-england-healthcare-system)

2. As a CAMO user, go to the VHA CAMO queue and make your way to the Case Details for one of the appeals. Select "Send to Board Intake" from the actions dropdown. Select a radio option and enter text into the textarea field. Submit the modal, navigate back to the Case Details page, and review the case timeline.

3. As the CAMO user, go to the Case Details page for **five** appeals and select "Assign to Program Office". Send all of them to the "Community Care - Payment Operations Management" office.

4. As a VHA PO user, go to the VHA PO queue (listed in step 1). For one of the appeals, go to the Case Details page and select "Return to VHA CAMO team". Enter text into the textarea field and submit. Review the resulting case timeline on the case details page.

5. As a VHA PO user, go to the Case Details page for another appeal and this time select "Documents ready for VHA CAMO team review" from the dropdown. Select the location of the documents (besides "Other") and enter text into the text box. Navigate back to the Case Details page and review the "Task Instructions" entry in the case timeline.

6. Repeat step 5 for another appeal. This time select "Other" and fill in the additional textbox.

7. As a VHA PO user, go to the Case Details page for **two** appeals and select "Assign to VISN" for both. Assign both appeals to the "VA New England Healthcare System" VISN.

8. As a VISN user, go to the Case Details page for one of the appeals and select "Return to Program Office". Enter text into the textarea field and submit. Review the resultant case timeline entry.

9. As a VISN user, go to the Case Details page for another appeal. This time select "Documents ready for VHA CAMO team review" (this will be "Documents ready for VHA Program Office team review" once 9206 is merged in). Enter location of documents, text into the textarea field, and then submit. Observe the case timeline entry.

10. As a VHA PO user, send the appeal from step 9 to VHA CAMO as ready for review. Review the case timeline.

11. As a VHA CAMO user, send the appeals from steps 5,6, and 10 to BVA Intake with the "Send to Board Intake" task action. Review the timelines for all three apepals.

### Education Scenarios

1. Open two browser windows where one in a incognito/private browsing window. Log into one as an EMO user (ex: `EMOUSER`) and in the other log in as an EDU RPO user (ex: `EDURPOUSER`).

**For reference:**
[EMO Queue](http://localhost:3000/organizations/edu-emo)
[EDU RPO Queue](http://localhost:3000/organizations/buffalo-rpo)

2. As the EMO user, go to the EMO queue. Go to the Case Details page for an appeal. Select "Return to Board Intake" from the actions dropdown. Select a reason for return and enter text in the textarea field. Submit the modal. Navigate back to the Case Details page and review the "Task Instructions" entry in the case timeline.

3. Go to the Case Details page for another one of the appeals. Select "Documents ready for Board Intake review" from the actions dropdown. Select the location of the documents (besides "Other") and enter text into the text box. Navigate back to the Case Details page and review the "Task Instructions" entry in the case timeline.

4. Repeat step 3 for another appeal. This time select "Other" and fill in the additional textbox.

5. Return once again to the EMO queue. Go to the Case Details page for another appeal and select "Assign to Regional Processing Office". Assign the appeal to the Buffalo RPO. Enter some text into the textarea field. **Do this for two more appeals.**

6. As the EDU RPO user go to the Buffalo RPO's queue (linked above under step 1). Go to the Case Details page for one of the appeals. Select "Return to Executive Management Office". Enter text into the textarea field and submit the modal. Go back to the Case Details page and observe the case timeline.

7. As the EDU RPO user go to another appeal's Case Details page and select "Documents ready for Board Intake review". Select the location of the documents (besides "Other") and enter text into the text box. Navigate back to the Case Details page and review the "Task Instructions" entry in the case timeline.

8. Repeat step 7 for another appeal. This time select "Other" and fill in the additional textbox.

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

## User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

### VHA Caregiver Support Program
 SCENARIO|BEFORE|AFTER
 ---|---|---
 VHA CSP sends appeal to BVA Intake as ready for review|pre-fix-csp-to-intake-r4r|csp-to-intake-r4r
 VHA CSP returns appeal to BVA Intake due to an error||csp-to-intake-return-error

 ### VHA CAMO/VHA PO/VISN
 SCENARIO|BEFORE|AFTER
 ---|---|---
 CAMO sends appeal back to BVA Intake||camo-to-bva-intake
 CAMO sends appeal to VHA PO. The PO returns it.||camo-to-po-they-cancel
 CAMO sends appeal to VHA PO. They mark it as ready for review.||camo-to-po-then-back-as-r4r
 CAMO sends appeal to VHA PO. They mark it as ready for review. CAMO sends it to BVA Intake||camo-to-po-to-camo-to-inta
 CAMO sends appeal to VHA PO. They assign it to a VISN. The VISN returns the appeal to PO.|po-to-visn-to-po-cancel|po-to-visn-they-cancel
 CAMO sends appeal to VHA PO. They assign it to a VISN. The VISN marks at as ready for review.|po-to-visn-to-po-r4r|po-to-visn-to-po-as-r4r
 CAMO sends appeal to VHA PO. They assign it to a VISN. The VISN marks at as ready for review. The PO marks it ready for review. CAMO marks it as ready for review.|Screenshot from 2022-09-28 09-04-19|camo-to-po-to-visn-to-po-to-camo-to-intake-r4r

 ### Education EMO/Education RPO
 SCENARIO|BEFORE|AFTER
 ---|---|---
 EMO to BVA Intake as ready for review||emo-to-intake
 EMO return to BVA Intake due to error||emo-return-to-intake
 EMO sends appeal to RPO, and the RPO returns it to the EMO||emo-to-rpo-and-rpo-cancels
 EMO to RPO, then RPO to BVA Intake as ready for review||emo-to-rpo-to-intake

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### CODE REVIEW

**Review by TylerBroyles — CHANGES_REQUESTED** [review]
Nice job! I added a couple of possible code changes that might be worth changing depending on the context.

File: client/app/queue/utils.js [hunk 1]
```diff
@@ -869,3 +869,47 @@ export const statusLabel = (appeal) => {
...
+    return task.parentId === parseInt(parentTask.taskId, 10);
+  });
+};
+
+export const getPreviousTaskInstructions = (parentTask, tasks) => {
+  let reviewNotes = null;
+
+  const childTask = getMostRecentChildTask(parentTask, tasks);
+
+  if (childTask && childTask.instructions?.[1] && childTask.instructions.length > 0) {
```
[the remark this answers is not in the document]
Comment (ThorntonMatthew) [reply client/app/queue/utils.js]: Thank you! I think this is a great idea.

File: client/app/queue/utils.js [hunk 2]
```diff
@@ -869,3 +869,47 @@ export const statusLabel = (appeal) => {
...
+      break;
+    case 'EducationRpo':
+      reviewNotes = 'Regional Processing Office';
+      break;
+    default:
+      break;
+    }
+  }
+
+  const previousInstructions = reviewNotes ? childTask.instructions?.[1] : null;
```
[the remark this answers is not in the document]
Comment (ThorntonMatthew) [reply client/app/queue/utils.js]: It just needs to be falsey, so this will work. Thanks!

**Review by TylerBroyles — APPROVED** [review]
Great job! Looks good to me.

**Review by TylerBroyles — APPROVED** [review]
Looks good to me.

**Pull request merged**